floodlight/io/dfl.py

```diff
@@ -539,55 +613,77 @@ def read_event_data_xml(
+        Teamsheet-object for the home team used to create link dictionaries of the form
+        `links[team][jID] = xID` and  `links[team][pID] = jID`. The links are used to
+        map players to a specific xID in the respective XY objects. Should be supplied
+        if that order matters. If given as None (default), teamsheet is extracted from
+        the Match Information XML file.
+    away_teamsheet: Teamsheet, optional
+        Teamsheet-object for the away team. If given as None (default), teamsheet is
         extracted from the Match Information XML file.
-    links_pID_to_jID: Dict, optional
-        A link dictionary of the form `links[team][pID] = jID` where pID is the PersonId
-        specified in the DFL Match Information file.
 
     Returns
     -------
-    data_objects: Tuple[XY, XY, XY, XY, XY, XY, Code, Code, Code, Code, Pitch]
-        XY-, Code-, and Pitch-objects for both teams and both halves. The order is
-        (home_ht1, home_ht2, away_ht1, away_ht2, ball_ht1, ball_ht2,
-        possession_ht1, possession_ht2, ballstatus_ht1, ballstatus_ht2, pitch)
-
+    data_objects: Tuple[XY, XY, XY, XY, XY, XY, Code, Code, Code, Code, Pitch,
+                        Teamsheet, Teamsheet]
+        XY-, Code-, Pitch-, and Teamsheet-objects for both teams and both halves. The
+        order is (home_ht1, home_ht2, away_ht1, away_ht2, ball_ht1, ball_ht2,
+        possession_ht1, possession_ht2, ballstatus_ht1, ballstatus_ht2, pitch,
+        home_teamsheet, away_teamsheet).
     """
     # read metadata
     pitch = read_pitch_from_mat_info_xml(filepath_mat_info)
 
-    # create or check links
-    if links_jID_to_xID is None or links_pID_to_jID is None:
-        links_jID_to_xID, links_pID_to_jID = create_links_from_mat_info(
-            filepath_mat_info
-        )
+    # create or check teamsheet objects
+    if home_teamsheet is None and away_teamsheet is None:
+        teamsheets = read_teamsheets_from_mat_info_xml(filepath_mat_info)
+        home_teamsheet = teamsheets["Home"]
+        away_teamsheet = teamsheets["Away"]
+    elif home_teamsheet is None:
+        teamsheets = read_teamsheets_from_mat_info_xml(filepath_mat_info)
+        home_teamsheet = teamsheets["Home"]
+    elif away_teamsheet is None:
+        teamsheets = read_teamsheets_from_mat_info_xml(filepath_mat_info)
+        away_teamsheet = teamsheets["Away"]
     else:
         pass
         # potential check
 
+    # create links
+    if "xID" not in home_teamsheet.teamsheet.columns:
+        home_teamsheet.add_xIDs()
+    if "xID" not in away_teamsheet.teamsheet.columns:
+        away_teamsheet.add_xIDs()
+    links_jID_to_xID = {}
+    links_jID_to_xID["Home"] = home_teamsheet.get_links("jID", "xID")
+    links_jID_to_xID["Away"] = away_teamsheet.get_links("jID", "xID")
+    links_pID_to_jID = {}
+    links_pID_to_jID["Home"] = home_teamsheet.get_links("pID", "jID")
+    links_pID_to_jID["Away"] = away_teamsheet.get_links("pID", "jID")
```

Comment:
Could be assigned during dict creation:
links_jID_to_xID = {
   "Home": = home.teamsheet.get_links .... ,
   "Away": ...
}